server/models/User.js

```diff
@@ -48,10 +59,40 @@ export default class User {
     return Promise.resolve(this.strip());
   }
 
+  async login() {
+    try {
+      const authQuery = `SELECT * FROM authentication WHERE username = '${this.username}'`;
+      const getAuth = await client.query(authQuery);
+      if (getAuth.rows.length === 0) return new Error('user not found');
+      this.username = getAuth.rows[0].username;
+      this.authId = getAuth.rows[0].id;
+      const correctPassword = await bcrypt.compare(this.password, getAuth.rows[0].password);
+      if (correctPassword) {
+        const userQuery = `SELECT * FROM users WHERE "authId"=${this.authId}`;
+        const getUser = await client.query(userQuery);
+        this.id = getUser.rows[0].id;
+        this.email = getUser.rows[0].email;
+        this.fullName = getUser.rows[0].fullName;
+        this.token = this.genToken();
+        return Promise.resolve(this.strip());
+      }
+      throw new Error('user not found');
+    } catch (error) {
+      return Promise.reject(error);
+    }
+  }
+
+  genToken() {
```

Comment:
Try using the asynchronous version of the jwt.sign (Add a callback)
```javascript
getToken(){
  return new Promise((resolve,reject) =>{
    jwt.sign({
       exp: (Math.floor(Date.now() / 1000) + (60 * 60)) * 24 * 7,
       data: this.strip(),
     }, process.env.JWT_SECRET, (error, signedToken)=>{
        if (error){
          reject(new Error(error.message));
        }else{
          resolve(signedToken);
        }
  })
})
}
```

Then call the asynchronous version of this function as well

Reply:
Ok, I'll do try this